tot/df_utils.py

```diff
@@ -948,13 +1043,18 @@ def unfold_dict_of_folds(folds_dict, k):
             training data
 
             validation data
+    Raises
+    -------
+        ValueError
+            If number of folds in folds_dict does not correspond to k.
     """
     folds = []
     df_train = pd.DataFrame()
     df_test = pd.DataFrame()
     for j in range(0, k):
         for key in folds_dict:
-            assert k == len(folds_dict[key])
+            if k != len(folds_dict[key]):
+                raise ValueError("Number of folds in folds_dict does not correspond to k")
```

Comment:
@ankke Sounds god to me for now. We are doing lots of functional programming here. Here the convention for using `_` is not as clear as in class-based programming. Unfortunately, we did not commit on a clear convention for using `_` in this project yet. Since the `df_utils.py` only is supposed to contain functions that are not used by the user, we could either go for using `_` on all functions or not at all. 
Happy to discuss the details, hear your expertise on it and find a good convention there.